src/ImageSharp/Common/Helpers/GaussianEliminationSolver.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+using System.Numerics;
+
+namespace SixLabors.ImageSharp.Common.Helpers;
+
+/// <summary>
+/// Represents a solver for systems of linear equations using the Gaussian Elimination method.
+/// This class applies Gaussian Elimination to transform the matrix into row echelon form and then performs back substitution to find the solution vector.
+/// This implementation is based on: https://www.algorithm-archive.org/contents/gaussian_elimination/gaussian_elimination.html
+/// </summary>
+/// <typeparam name="TNumber">The type of numbers used in the matrix and solution vector. Must implement the <see cref="INumber{TNumber}"/> interface.</typeparam>
+internal static class GaussianEliminationSolver<TNumber>
+    where TNumber : INumber<TNumber>
```

Comment:
I think we should be looking for other opportunities to use `INumber<TSelf>` etc in the code to reduce duplication as we slowly modernize this however, for this internal type will we ever not use `float`?

Reply:
When I started I did not know if it was only float or not, so I tried with `INumber`. It was fun to discover those new things :)
I don't mind replacing it with `float`. I'll let you decide this, just let me know if this stay this way or if we change. (I don't know yet if ImageSharp is using double/decimal somewhere, from what I saw it was mainly `float` and `int`.

I also tried when, I was doing some benchmark, to replace TNumber with float, and the performances were the same.

Reply:
ImageSharp uses `float` everywhere as we use `Vector4` as a pixel interop. I think using float here would be best.

Reply:
Ok, I'll change that